source/MRMesh/MR2DContoursTriangulation.cpp

```diff
@@ -431,18 +440,38 @@ void PlanarTriangulator::resolveIntersectios_()
         // update queue
         queue_.push( ComaparableVertId{ &mesh_,vInter } );
     }
+    return true;
+}
+
+Mesh triangulateContours( const Contours2d& contours, bool mergeClosePoints /*= true*/ )
+{
+    PlanarTriangulator triangulator( contours, mergeClosePoints, false );
+    if ( auto res = triangulator.run() )
+        return *res;
+    else
+        return Mesh();
+}
+
+Mesh triangulateContours( const Contours2f& contours, bool mergeClosePoints /*= true*/ )
+{
+    const auto contsd = copyContours<Contours2d>( contours );
+    PlanarTriangulator triangulator( contsd, mergeClosePoints, false );
+    if ( auto res = triangulator.run() )
+        return *res;
```

Comment:
```
return std::move( *res );
```
otherwise you will get extra mesh copy: https://gcc.godbolt.org/z/7849c949c

Reply:
fixed